pallets/avn-anchor/src/lib.rs

```diff
@@ -288,9 +336,57 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::set_checkpoint_fee())]
+        #[pallet::call_index(6)]
+        pub fn set_checkpoint_fee(
+            origin: OriginFor<T>,
+            chain_id: ChainId,
+            new_fee: BalanceOf<T>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            CheckpointFee::<T>::insert(chain_id, new_fee);
+            Self::deposit_event(Event::CheckpointFeeUpdated { chain_id, new_fee });
+
+            Ok(())
+        }
     }
 
     impl<T: Config> Pallet<T> {
+        pub(crate) fn charge_fee(handler: T::AccountId, chain_id: ChainId) -> DispatchResult {
+            let payment_nonce = PaymentNonces::<T>::get(&handler);
+            let checkpoint_fee = Self::get_checkpoint_fee(chain_id);
+            let treasury_account_id = compute_treasury_account_id::<T>();
+
+            T::FeeHandler::pay_fee(
```

Comment:
I think we should have a function called `pay_treasury` in `FeeHander` and avoid passing in a `TreasuryAccount` and deriving an accountID from it. In case we ever change how we derive that account in TokenManager this would diverge without us knowing about it.